Translated more references to Hindi

ayushjainrksh opened the pull request.

limzykenneth commented:
Are you thinking of adding more entries to the reference translation? If so you can add a "WIP" to your PR title or even put this PR in draft mode, then we'll merge when you are ready. 

Also it will be very helpful if you can make your PR title a bit more descriptive, saying this is about translation of the reference into Hindi, otherwise it could look like you are adding new reference entries to the original reference. Thanks.

ayushjainrksh commented:
@limzykenneth thanks for the review. I will be converting all the references to Hindi. I can't seem to find the references file in English ( en.json). I'll also work on your suggestion.

limzykenneth commented:
There isn't an English version of the file for reference translation. You'll basically need to copy the entries from the other files manually.

ayushjainrksh commented:
Okay, @limzykenneth thanks for letting me know. I will use es.json for the translation.

ayushjainrksh commented:
@limzykenneth @lmccart I have updated the list of translations and added `"hi"` in `package.json, index.hbs(reference), i18n.hbs`. I have also translated the index page of the references to Hindi. Now I just need to add translations for the functions in reference. Please let me know if I am doing something wrong.

ayushjainrksh commented:
@limzykenneth @lmccart  The translation of the `Color` functions in the reference to Hindi has been added. As I have completed the translation of the index page and a section of the functions in the reference, it would be great if someone could review my work so that I can continue ahead. It would be better if the mentor has a knowledge of Hindi (such as Shaharyar). I used a small script to print out the translation to the terminal and then manually checked if each line has been translated correctly.

shaharyar-shamshi commented:
@ayushjainrksh @limzykenneth it translation is working fine

limzykenneth commented:
Nothing stood out for me as problematic. You should build the website and preview it in the browser using `grunt build` to check though.

ayushjainrksh commented:
Thanks for the review @shaharyar-shamshi . 
@limzykenneth I did check the translation of the website in the browser. In fact I have set up the reference translation like all other translations in production. You can check in my previous commits that I have added `hi` to i18n references too.
![hiTrans](https://user-images.githubusercontent.com/33171576/75990513-03c1a700-5f1b-11ea-8759-1058764681d9.png)

limzykenneth commented:
I have a new PR that removes the need to add those additional stuff but don't worry about it too much for now, I'll take care of merging that into the translation branch after the PR has been merged.

ayushjainrksh commented:
Okay thanks for sharing. I will work on the translations in the current branch.

limzykenneth commented:
I'm not involve with the GSoC process so I'm not sure if my advice would be helpful at all. You are still very welcomed to share your draft proposal on the forum where you can get advice from a wide variety of audience including past GSoC scholar. I can't guarantee my availability to look at a proposal but you are welcomed to tag me on the forum if you do share it.

ayushjainrksh commented:
Yeah I'll ask for help on forum. Thanks for suggestion :)

ayushjainrksh commented:
@limzykenneth @lmccart I have translated a few more functions in my latest commit using the output from a small script I wrote. It is working perfectly in my opinion. Can you please look at my latest commits and let me know if it is good enough. I am also wondering if it is the right time to merge the translations.

limzykenneth commented:
Nothing looked out of the ordinary for me. We can merge whenever you are ready, as long as you have reached a reasonable break point, we can merge this then if you like continue on a different PR.

ayushjainrksh commented:
Okay thanks @limzykenneth . I have tested the translations and they stood out great for me. @shaharyar-shamshi Can you check if the translations are grammatically correct and are free of errors. I think we can merge this PR now.

shaharyar-shamshi commented:
@ayushjainrksh I have deployed locally it is working great there is no grammatical error @limzykenneth it can be merged

ayushjainrksh commented:
Great! I will update the WIP status then. @limzykenneth please merge the PR when you have time.

limzykenneth commented:
Good job, thanks!

davepagurek commented on src/layouts/ReferenceItemLayout.astro:
```diff
@@ -169,6 +171,23 @@ const seenParams: Record<string, true> = {};
+        (entry.data.return) && (
+          <>
+            <div class="mb-xl">
+              <h2 class="mb-md text-h3">{t("Return")}</h2>
```
Looks like on the old site it was "Returns" with an s, maybe we can do the same here too?

<img width="905" alt="image" src="https://github.com/user-attachments/assets/170cb3f6-45cd-4600-b1b9-ab3b452b11cc">


```suggestion
              <h2 class="mb-md text-h3">{t("Returns")}</h2>
```

shibomb replied:
@davepagurek 
Thanks for reply.

This is a suggestion for improvement.

Indeed, the old version was with "s".
I assume this “s” means plural, but can the returned value be plural?

- It is certainly possible to have multiple meanings by using arrays or Json, but I think that should be explained in the Description.

- As a property of the .mdx file is “return:”.

- Also, for example, there is no “s” in processing.org.
  (I haven't checked all the pages... For example, https://processing.org/reference/cos_.html .)

Please consider the above!

If you need the “s”, I will modify the necessary part :-)

davepagurek replied:
Returns with an s is I think more common, e.g. [Tone.js](https://tonejs.github.io/docs/15.0.4/classes/Context.html#createWaveShaper), [React](https://react.dev/reference/react/useCallback#returns), [matter.js](https://brm.io/matter-js/docs/classes/Constraint.html), and [lodash](https://lodash.com/docs/4.17.15#now). I think the reason for it isn't to consider "return" as a plural noun, but rather as a verb, e.g. "this function returns a number", or just "returns: number" for short in this case. In this case the s just means present tense, "it returns" as opposed to "it returned" in the past.

shibomb replied:
I understand.
I will correct it!

shibomb replied:
@davepagurek 
I have updated the branch.
Please re-review.
Thank you.

davepagurek reviewed (commented):
Thanks, looks good! I just noticed one difference from before, but otherwise good to go.

davepagurek reviewed (approved):
Thanks, looks great!
